src/commands/Moderation/multiban.ts

```diff
@@ -28,93 +28,103 @@ export default class extends Command {
 	}
 
 	public async run(context: CommandContext, args: string[]): Promise<void | Message> {
-		const targets: RegExpMatchArray = args.slice(0).join(" ").match(UserMentionAndID) ?? [];
-		let reason: string = args.slice(targets.length).join(" ").replace(UserMentionAndID, "");
+		const potentialTargets: RegExpMatchArray = <RegExpMatchArray>args.slice(0).join(" ").match(UserMentionAndID);
+		let validTargets: (GuildMember | User)[] = [];
+		let invalidTargets: number = 0;
+		let fullList: string = "";
+		let reason: string = args.slice(potentialTargets?.length).join(" ").replace(UserMentionAndID, "");
 
 		if (reason === "") reason = await this.client.bulbutils.translate("global_no_reason", context.guild?.id, {});
-		let fullList: string = "";
 
-		if (targets.length <= 1) {
+		for (const potentialTarget of potentialTargets) {
+			const t = potentialTarget.replace(NonDigits, "");
+			if (!t.length) continue;
+
+			let target: User | GuildMember | null = t ? <GuildMember>await context.guild?.members.fetch(t).catch(() => null) : null;
+
+			if (!target) {
+				try {
+					target = await this.client.users.fetch(t);
+				} catch (error) {
+					invalidTargets++;
+					continue;
+				}
+			}
+
+			if (target instanceof GuildMember)
+				if (await this.client.bulbutils.resolveUserHandle(context, await this.client.bulbutils.checkUser(context, target), target.user)) continue;
+
+			validTargets = [...validTargets, target];
+		}
+
+		if (validTargets.length === 1) {
 			await context.channel.send(
 				await this.client.bulbutils.translate("action_multi_less_than_2", context.guild?.id, {
 					action: await this.client.bulbutils.translate("action_multi_types.ban", context.guild?.id, {}),
 				}),
 			);
-			return await this.client.commands.get("ban")!.run(context, args);
+			return this.client.commands.get("ban")!.run(context, [validTargets[0].id, ...reason.split(" ")]);
 		}
 
-		context.channel.send(await this.client.bulbutils.translate("global_loading", context.guild?.id, {})).then(msg => {
-			setTimeout(() => msg.delete(), (args.length - 0.5) * massCommandSleep);
-		});
-
-		for (let i = 0; i < targets.length; i++) {
-			if (targets[i] === undefined) continue;
-			await this.client.bulbutils.sleep(massCommandSleep);
-
-			const t: Snowflake = targets[i].replace(NonDigits, "");
-			let infID: number;
-			let target: any = await context.guild?.members.cache.get(t);
-			const notInGuild: boolean = !target;
-
-			if (!notInGuild) {
-				if (await this.client.bulbutils.resolveUserHandle(context, await this.client.bulbutils.checkUser(context, target), target.user)) return;
-			}
+		await context.channel.send(await this.client.bulbutils.translate("global_loading", context.guild?.id, {})).then(msg => {
+			setTimeout(async () => {
```

Comment:
Errors thrown in structures like `.then` and `setTimeout` will either not be logged in our error logging, or will lose the line information that tells us where to look in our code

Reply:
Fixed in 4216fa7